(SERVER-2215) Update cljs tools and use figwheel-sidecar

lazartea opened the pull request:
cljs-dashboard-widgets used older clojure, clojurescript, and cljsjs/d3 dependencies. I updated these to modern versions. Figwheel required a lein dependency to run `lein figwheel`, but upgrading to figwheel-sidecar allowed me to remove that dependency.

lazartea committed:
(Maint) Update releases and snapshots repositories

The old repositories URLS were incorrect, which caused an error message when trying to start a REPL. The REPL now starts successfully.

justinstoller commented on project.clj:
```diff
@@ -6,13 +6,13 @@
 
                  ;; transitive dependencies
-                 [org.clojure/clojurescript "1.7.228"]
+                 [org.clojure/clojurescript "1.10.238" :exclusions [org.clojure/clojure com.fasterxml.jackson.core/jackson-core]]
```
Few things:
  * Long lines should be broken up. With at the artifact name one per line here, potentially the `:exclusions` keyword starting a newline (eg https://github.com/puppetlabs/puppetserver/blob/master/project.clj#L174-L178).
  * We tend to prefer explicit conflict resolution of transitive dependencies over exclusions, that's not always possible, but we should probably do that with jackson at least.
 * Looking at these dependencies lists it might be worth while to pull in [clj-parent](https://github.com/puppetlabs/clj-parent) (see puppetserver's usage of clj-parent for an example, reading lein's [docs](https://github.com/technomancy/leiningen/blob/master/doc/MANAGED_DEPS.md) is probably a good first step).

lazartea committed:
(SERVER-2215) Update cljs tools and use figwheel-sidecar

cljs-dashboard-widgets used older clojure, clojurescript, and cljsjs/d3 dependencies. I updated these to modern versions and added clj-parent to manage the rest of the dependencies in the project. Upgrading to d3 v4 broke some of the metrix-box animation, so I fixed that as well. Figwheel required a lein dependency to run lein figwheel, but upgrading to figwheel-sidecar allowed me to remove that dependency.

lazartea commented:
Superseded by #9
